docs/tutorial.rst

```diff
@@ -863,6 +863,167 @@ not ancestral to these census nodes.
+                         (equivalent to migration from 0 to 1 forwards in time)
+    >>> print(parsed_tuple[1][1])
+    Mass migration: Lineages moved with probability 1.0 backwards in time with source 2 & dest 0
+                         (equivalent to migration from 0 to 2 forwards in time)
+    >>> print(parsed_tuple[1][2])
+    Mass migration: Lineages moved with probability 1.0 backwards in time with source 3 & dest 0
+                         (equivalent to migration from 0 to 3 forwards in time)
+
+The above output indicates that --- viewed backwards in time --- lineages from
+populations 1, 2, and 3 are consecutively moved into population 0. Viewed forwards in
+time instead, this means that population 3 is the first to diverge, followed by
+population 2 and finally the divergence between populations 0 and 1. This
+sequence of divergences corresponds to the species tree if population 3 is orangutan
+and populations 2, 1, and 0 are gorilla, chimpanzee, and human, respectively. While
+the parsed species names are not used as population labels, they are included in the
+population configurations in the form of metadata, with the "species_name" tag:
+
+.. code-block:: python
+
+    >>> print(parsed_tuple[0][0].metadata)
+    {'species_name': 'human'}
+    >>> print(parsed_tuple[0][1].metadata)
+    {'species_name': 'chimpanzee'}
+    >>> print(parsed_tuple[0][2].metadata)
+    {'species_name': 'gorilla'}
+    >>> print(parsed_tuple[0][3].metadata)
+    {'species_name': 'orangutan'}
+
+As the above output shows, the information on the topology of the species tree is
+fully included in the set of population configurations and mass migration events.
+It also illustrates that it is always the left one (when viewed from the root
+towards the tips of the species tree in the tree figure above) of the two populations
+descending from a divergence event that is used as the destination population in
+mass migration events.
+
+The population configurations also define the population size:
+
+.. code-block:: python
+
+    >>> print(parsed_tuple[0][0].initial_size)
+    10000.0
+    >>> print(parsed_tuple[0][1].initial_size)
+    10000.0
+    >>> print(parsed_tuple[0][2].initial_size)
+    10000.0
+    >>> print(parsed_tuple[0][3].initial_size)
+    10000.0
+
+The population size is 10,000 because this value was specified for Ne when
+calling :func:`.parse_species_tree`. The growth rates are zero for all populations,
+meaning that they all have constant population sizes:
+
+.. code-block:: python
+
+    >>> print(parsed_tuple[0][0].growth_rate)
+    0.0
+    >>> print(parsed_tuple[0][1].growth_rate)
+    0.0
+    >>> print(parsed_tuple[0][2].growth_rate)
+    0.0
+    >>> print(parsed_tuple[0][3].growth_rate)
+    0.0
+
+To simulate under the model corresponding to the species tree, the population
+configurations and mass migration events are used as input for
+:func:`.simulate`:
+
+.. code-block:: python
+
+    >>> for pop_conf in parsed_tuple[0]:
```

Comment:
I think it would be better to use the ``samples`` argument here rather than updating the pop_configs. Something like

```python
samples = [msprime.Sample(population=0) for _ in range(2)] + [
     [msprime.Sample(population=1) for _ in range(2)]
tree_sequence = msprime.simulate(
    samples=samples,
    population_configurations=pop_configs,
    # etc)
```

Reply:
I wouldn't mind to rewrite it this way, but the code doesn't seem to work:
```
samples = [[msprime.Sample(population=0) for _ in range(2)] + [msprime.Sample(population=1) for _ in range(2)]]
TypeError: __new__() missing 1 required positional argument: 'time'
```

```
samples = [[msprime.Sample(population=0, time=0) for _ in range(2)] + [msprime.Sample(population=1, time=0) for _ in range(2)]]
tree_sequence = msprime.simulate(
         samples=samples,
         population_configurations=pop_configs,
         demographic_events=demographic_events)
ValueError: Sample size must be >= 2
```

```
samples = [[msprime.Sample(population=0, time=0) for _ in range(2)] +
     [msprime.Sample(population=1, time=0) for _ in range(2)] +
     [msprime.Sample(population=2, time=0) for _ in range(2)] +
     [msprime.Sample(population=3, time=0) for _ in range(2)]]
tree_sequence = msprime.simulate(
         samples=samples,
         population_configurations=pop_configs,
         demographic_events=demographic_events)
ValueError: Sample size must be >= 2